nengo_gui/components/htmlview.py

```diff
@@ -10,31 +12,50 @@ class HTMLView(Component):
 
     def __init__(self, obj):
         super(HTMLView, self).__init__()
-        self.obj = obj
-        self.obj_output = obj.output
         self.data = collections.deque()
 
+        self.obj = obj
+        self.obj_output = self.obj.output
+        self.callable_html = callable(obj.output._nengo_html_)
+        if self.callable_html:
```

Comment:
I agree. Maybe provide a helper function that automatically instantiates the right class. Something like
```python
def HTMLView(obj):
    if callable(obj.output._nengo_htlm):
        return CallableHTMLView(obj)
    else
        return StaticHTMLView(obj)
```